compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptions.kt

```diff
@@ -0,0 +1,317 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.bottlerocketstudios.compose.R
+
+data class DevOptionsState(
+    val environmentNames: State<List<String>>,
+    val environmentSpinnerPosition: State<Int>,
+    val baseUrl: State<String>,
+    val appVersionName: String,
+    val appVersionCode: String,
+    val appId: String,
+    val buildIdentifier: String,
+    val onEnvironmentChanged: (Int) -> Unit,
+    val onRestartCtaClick: () -> Unit,
+    val onForceCrashCtaClicked: () -> Unit
+)
+
+@Composable
+fun DevOptionsScreen(state: DevOptionsState) {
+    Scaffold(floatingActionButton = { FabLayout(state.onRestartCtaClick) }) {
+        Column(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            DropdownEnvMenu(state)
+            ScrollContent(state)
+        }
+    }
+}
+
+@Composable
+private fun FabLayout(onFabClick: () -> Unit) {
+    FloatingActionButton(
+        onClick = { onFabClick() },
+        backgroundColor = MaterialTheme.colors.secondary,
+        content = {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_restart_24),
+                contentDescription = "Restart",
+                tint = Color.White
+            )
+        }
+    )
+}
+
+@Composable
+private fun ScrollContent(state: DevOptionsState) {
+    Column {
+        LazyColumn(
+            modifier = Modifier
+                .padding(start = 12.dp, end = 12.dp)
+                .fillMaxHeight()
+                .fillMaxWidth()
+                .weight(1F)
+        ) {
+            item {
+                CardDivider(Color.Transparent)
+                CardLayout { EnvironmentCard(baseUrl = state.baseUrl.value) }
+                CardDivider(MaterialTheme.colors.primary)
+            }
+            item {
+                CardLayout { AppInfoCard(state = state) }
+                CardDivider(MaterialTheme.colors.primary)
+            }
+            item {
+                CardLayout { MiscFunctionalityCard(onForceCrashCtaClick = state.onForceCrashCtaClicked) }
+                CardDivider(Color.Transparent)
+            }
+        }
+    }
+}
+
+@Composable
+private fun BottomButton(buttonText: String, onClick: () -> Unit) {
+    Button(
+        onClick = { onClick() },
+        modifier = Modifier
+            .padding(8.dp)
+            .wrapContentHeight()
+            .fillMaxWidth(),
+        colors = ButtonDefaults.textButtonColors(
+            backgroundColor = MaterialTheme.colors.primary
+        ),
+        shape = RoundedCornerShape(7.dp),
+        contentPadding = PaddingValues(12.dp)
+    ) {
+        Text(
+            text = buttonText,
+            style = TextStyle(color = Color.White, fontSize = 14.sp)
+        )
+    }
+}
+
+@Composable
+private fun EnvironmentCard(baseUrl: String) {
+    CardTitle(cardTitle = "Selected Environment Info")
+    TitleValueRow(title = "Base URL", entryValue = baseUrl)
```

Comment:
Translations?  Could be a candidate for another PR.

Reply:
Yeah these should be updated to string resources just for consistency, though since its a dev only screen translations aren't a massive deal imo.

Reply:
One of the goals of the dev options screen was to be self containing since it is one of the few UI screens that are kept when starting a new project (leads might only want to keep the splash and possibly a slimmed down login screen). Since it should never be accessible on production builds (and therefore only accessible to Devs/QA on internal builds), hardcoded strings were used.

That being said I agree with following the best practice to add the strings to xml string resources to showcase proper usage patterns. If/when that happens, I think we should do one of the following to separate/highlight the strings for that screen:

1. Use a separate `dev_options_strings.xml` (or `strings_dev_options.xml` 🤷🏻 ) with the strings OR
2. Use a clearly marked section for DevOptions strings in the main strings.xml

Thoughts?

Reply:
Hmm with the goal being a standalone screen, I wonder if it makes sense to pull all dev options stuff into its own module with the goal of it eventually being a versioned library. @BottleRocket-Colin and I were talking about the benefits of demo code living as libraries and the ability to silo code like this as well as version it for projects that use it seems beneficial. May need to be added to the end of the todo list though.